_rules/letter-spacing-not-important-24afc2.md

```diff
@@ -0,0 +1,160 @@
+---
+id: 24afc2
+name: Letter spacing in `style` attributes is not `!important`
+rule_type: atomic
+description: |
+  This rule checks that the letter spacing can be adjusted by the user, i.e. that `!important` is not applied to the style, unless the style has already exceeded the specified metric for retaining content visibility and functionality.
+accessibility_requirements:
+  wcag21:1.4.12: # Text Spacing (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Jey Nandakumar
+---
+
+## Applicability
+
+This rule applies to any HTML element that is [visible][], for which the `style` attribute [declares][declared] the [letter-spacing][] CSS property.
+
+## Expectation
+
+For the test target, one of the following is true:
+
+- the [declared][] value of its [letter-spacing][] property, specified via the `style` attribute, does not have the [important flag][]; or
+- the [computed][] value of its [letter-spacing][] property is at least `0.12` times its [computed][] [font-size][].
```

Comment:
I disagree with the way this is expressed, but as it seems most folks prefer this I'll just leave it for others to approve.

Reply:
It's literally in the specification:

> Each property declaration applied to an element contributes a declared value for that property associated with the element. 

https://drafts.csswg.org/css-cascade/#declared

Reply:
@WilcoFiers how would you rephrase this, noting the CSS spec definitions that @kasperisager refers to above?

Reply:
Here's an actual issue though: It's not clear how to handle `style` attributes where _multiple_ values are declared for the same property. Consider this:

```html
<p style="letter-spacing: 0.12; letter-spacing: 0.1;">
```

Such a `style` attribute doesn't have just _a_ declared value for the `letter-spacing` property, it has _several_. We want the last, valid declaration.

Reply:
Valid point @kasperisager. I think simply referencing to [cascaded value][https://www.w3.org/TR/css-cascade/#cascaded] instead of declared value for the first point in the expectation should solve for this. I have made the change accordingly.

Reply:
🤔 the cascaded value can come from somewhere else.
OTOH, not sure whether a the cascade can be won by something else when there is a declared value, so maybe this is indeed good…